### PR TITLE
Remove all of css/css-regions/

### LINKED ISSUE #21304: Remove css/css-regions?
@clopez noted that https://wpt.fyi/results/css/css-regions is almost entirely failing in all browsers.

Most of the tests were written a long time ago in https://github.com/w3c/csswg-test/commits/master/css-regions-1.

Is any browser engine team planning to implement CSS Regions? If not, we should remove these tests.

@astearns @rachelandrew you are listed in META.yml, what are your thoughts?

### PR DESCRIPTION
**Pull request by foolip**
Fixes https://github.com/web-platform-tests/wpt/issues/21304.

### CODE REVIEW

File: lint.whitelist [hunk 1]
```diff
@@ -475,9 +471,6 @@ CSS-COLLIDING-SUPPORT-NAME: css/*/LOCK
 # These are all the current "merge mismatch" errors the build system produces
 CSS-COLLIDING-SUPPORT-NAME: css/css-flexbox/support/200x200-green.png
 CSS-COLLIDING-SUPPORT-NAME: css/css-grid/grid-items/support/200x200-green.png
-CSS-COLLIDING-SUPPORT-NAME: css/css-scroll-snap/support/common.css
```
Comment (astearns) [inline lint.whitelist]: was this intentional?

File: lint.whitelist [hunk 2]
```diff
@@ -574,8 +563,6 @@ CSS-COLLIDING-REF-NAME: css/css-masking/clip-path/reference/clip-path-square-002
 
 # Duplicate filename where there's no actual merging
 # (i.e., we're being more aggressive than the build system at forbidding it)
-CSS-COLLIDING-REF-NAME: css/CSS2/positioning/position-relative-001-ref.xht
```
Comment (astearns) [inline lint.whitelist]: I see position-relative*.html deletions here, but not an xht one

File: lint.whitelist (hunk 1 above)
Comment (foolip) [reply lint.whitelist]: Yes, this lint lists all of the colliding names, and if only one was left after my initial edits I removed that one too. The lint still passes :)

File: lint.whitelist (hunk 2 above)
Comment (foolip) [reply lint.whitelist]: I didn't pay attention to the extension, but it appears as though the extension is ignored. I couldn't find where in the code the extension is stripped, but confirmed that the lint fails if these two lines are removed without making any other changes. Anyway, if the lint passes when a line of line.whitelist is removed, then that line can be removed :)

**Review by astearns — APPROVED** [review]

**Pull request merged**